flutter/lib/src/web/script_loader/web_script_dom_api.dart

```diff
@@ -0,0 +1,27 @@
+import 'dart:async';
+
+// ignore: depend_on_referenced_packages
+import 'package:web/web.dart';
+
+Future<void> loadScript(String src, String? integrity) {
+  final completer = Completer<void>();
+  final script = HTMLScriptElement()
+    ..src = src
+    ..crossOrigin = 'anonymous'
+    ..onLoad.listen((_) => completer.complete())
+    ..onError.listen((event) => completer.completeError('Failed to load $src'));
+
+  if (integrity != null) {
+    script.integrity = integrity;
+  }
+  // JS SDK needs to be loaded before everything else
+  final head = document.head;
+  if (head != null) {
+    if (head.hasChildNodes()) {
+      head.insertBefore(script, head.firstChild);
+    } else {
+      head.append(script);
+    }
+  }
```

Comment:
@lforst do you know if there is a risk in 'injecting' the script into the head like this, e.g being flagged by websites or anything like that? we do that that so we can use the JS SDK in Flutter Web

Reply:
I am actually not sure. The only thing I can think of (except for maybe lighthouse punishing you or whatever) is CSP potentially being a pain.

Reply:
ah got it, so ig it's preferrable that user's just manually add the script in their html file